pyomo/contrib/trustregion/examples/example2.py

```diff
@@ -0,0 +1,50 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+# This model is adapted from Noriyuki Yoshio's model for his and Biegler's
+# 2020 publication in AIChE
+
+from pyomo.environ import (
+    ConcreteModel, Var, Reals, ExternalFunction, sin, cos,
+    sqrt, Constraint, Objective)
+from pyomo.opt import SolverFactory
+
+m = ConcreteModel()
+m.name = 'Example 2: Yoshio'
+
+m.x1 = Var(initialize=0)
+m.x2 = Var(bounds=(-2.0, None), initialize=0)
+
+def ext_fcn(x, y):
+    return x**2 + y**2
+def grad_ext_fcn(args, fixed):
+    x, y = args[:2]
+    return [ 2*x, 2*y ]
+
+m.EF = ExternalFunction(ext_fcn, grad_ext_fcn)
+
+@m.Constraint()
+def con(m):
+    return 2*m.x1 + m.x2 + 10.0 == m.EF(m.x1, m.x2)
+
+m.obj = Objective(expr = (m.x1 - 1)**2 + (m.x2 - 3)**2 + m.EF(m.x1, m.x2)**2)
+
+def basis_rule(component, ef_expr):
+    x = ef_expr.arg(0)
+    y = ef_expr.arg(1)
+    return x**2 - y # This is the low fidelity model
```

Comment:
This comment is extremely pedantic so feel free to ignore it. The variables in this example vary from `x` and `y` to `x1` and `x2` depending on whether you are using the external function or the original model. It can be either confusing or revealing since the different variables' names can be used.
Moreover, there are a lot of warnings from CodeCov on this file, is this example not being used in regressions tests?

Reply:
@bernalde  - I updated both of the examples to use the same formatting and not mix/match `x`'s and `y`'s all over the place plus added tests for both example files.